packages/dropdowns/src/Dropdown/Dropdown.js

```diff
@@ -0,0 +1,185 @@
+    return {
+      onKeyDown: composeEventHandlers(onKeyDown, e => {
+        const PREVIOUS_KEY = rtl ? KEY_CODES.RIGHT : KEY_CODES.LEFT;
+        const NEXT_KEY = rtl ? KEY_CODES.LEFT : KEY_CODES.RIGHT;
+
+        if (downshift.isOpen) {
+          // Select highlighted item on TAB
+          if (e.keyCode === KEY_CODES.TAB) {
+            e.preventDefault();
+            e.stopPropagation();
+
+            downshift.selectHighlightedItem();
+          }
+
+          // Select previous item if available
+          if (
+            e.keyCode === PREVIOUS_KEY &&
+            previousIndexRef.current !== null &&
+            !downshift.inputValue
+          ) {
+            e.preventDefault();
+            e.stopPropagation();
+
+            downshift.selectItemAtIndex(previousIndexRef.current);
+          }
+
+          // Select current next item if available
+          if (e.keyCode === NEXT_KEY) {
+            const nextItemIndexes = Object.values(nextItemsHashRef.current);
+
+            if (nextItemIndexes.includes(downshift.highlightedIndex)) {
+              e.preventDefault();
+              e.stopPropagation();
+
+              downshift.selectItemAtIndex(downshift.highlightedIndex);
+            }
+          }
+        } else if (
+          (e.keyCode === KEY_CODES.ENTER || e.keyCode === KEY_CODES.SPACE) &&
+          !downshift.isOpen
+        ) {
+          e.preventDefault();
+          e.stopPropagation();
+
+          downshift.openMenu();
+        }
+      }),
+      ...other
+    };
+  };
+
+  const transformDownshift = ({ getInputProps, getToggleButtonProps, ...downshift }) => {
+    return {
+      getInputProps: p => getInputProps(customGetInputProps(p, downshift, isRtl(props))),
+      // The default aria-label provided by Downshift is invalid due to our DOM structure
+      getToggleButtonProps: p => getToggleButtonProps({ 'aria-label': undefined, ...p }),
+      ...downshift
+    };
+  };
+
+  return (
+    <Manager>
+      <Downshift
+        suppressRefError // https://github.com/downshift-js/downshift/issues/529 Allows us to provide props through context
+        isOpen={isOpen}
+        highlightedIndex={highlightedIndex}
+        selectedItem={selectedItem || null} // Ensures that selectedItem never becomes controlled internally by Downshift
+        inputValue={inputValue}
+        onStateChange={(changes, stateAndHelpers) => {
+          if (Object.prototype.hasOwnProperty.call(changes, 'isOpen')) {
+            onOpen && onOpen(changes.isOpen, stateAndHelpers);
+          }
+
+          if (Object.prototype.hasOwnProperty.call(changes, 'selectedItem')) {
+            if (selectedItems) {
+              const { itemToString } = stateAndHelpers;
+              const existingItemIndex = selectedItems.findIndex(item => {
+                return itemToString(item) === itemToString(changes.selectedItem);
+              });
+              const updatedSelectedItems = selectedItems.slice();
```

Comment:
`Array#from` would be clearer here that you want a new copy and not a reference.

Reply:
Neat. These both make shallow copies so it should be equivalent. Added.